File: 2019/Day01.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main(){

	ifstream inputFile ("Day01.txt");
	string line;

	int sum = 0;
	while(getline(inputFile, line)){

		int mass = stoi(line);

		do {
			int fuel = mass / 3 - 2;
			mass = max(fuel, 0);
			sum += mass;
		} while (mass > 0);
	}

	cout << sum << endl;
}

File: 2020/Day01.py
def main():

    sum = 2020

    numbers = []
    with open("Day01.txt", "r") as input_file:
        for line in input_file:
            numbers.append(int(line))

    numset = set()
    for number in numbers:
        if (sum - number) in numset:
            print(number * (sum - number))
            break
        else:
            numset.add(number)

main()